plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcPageSource.java

```diff
@@ -326,17 +333,19 @@ public String toString()
             implements ColumnAdaptation
     {
         private final int index;
+        private final UnaryOperator<Block> blockCoercer;
```

Comment:
Should we mention any additional information as a description ?

Reply:
It would be clearer to define a BlockTypeCoercion interface for `public Block apply(Block block)` and then use that in TypeCoercer instead of `Function<Block, Block>` and here instead of `UnaryOperator<Block>`.
It will also be easier to track possible implementations.

Reply:
The from and to types of the coercion would be useful info in the toString

Reply:
Instead of introducing another abstraction - We have created a new `ColumnAdaption` impl for coerced column.